Sources/Abstract/Wrapper.swift

```diff
@@ -0,0 +1,31 @@
+/*:
+# Wrapper
+
+And abstract definition for a type that wraps another, and guarantees both initialization with and access to the wrapped value.
+
+The requirement is that, by constructing a `Wrapper` with a certain `Wrapped`, we can retrieve exactly the same value, thus:
+
+Wrapped(object).unwrap == object
+
+In a sense it means that the function `Wrapper.init(_:)` is isomorphic: from an instance of `Wrapper` I can always get a `Wrapped` and viceversa.
```

Comment:
Yeah you're right, what I meant is that `Wrapper.init(_:)` and `Wrapper.unwrap` form an isomorphism, but it's probably better to more precisely say it in terms of `Wrapper` and `Wrapped`. I'm going to change it now and then merge.